test/test_transforms.py

```diff
@@ -1831,6 +1735,104 @@ def test_random_erasing(self):
+                aspect_ratio_obtained <= max(aspect_ratio_range) + epsilon) or
+               aspect_ratio_obtained == 1.0)
+        assert isinstance(i, int)
+        assert isinstance(j, int)
+        assert isinstance(h, int)
+        assert isinstance(w, int)
+
+
+def test_resize():
+
+    input_sizes = [
+        # height, width
+        # square image
+        (28, 28),
+        (27, 27),
+        # rectangular image: h < w
+        (28, 34),
+        (29, 35),
+        # rectangular image: h > w
+        (34, 28),
+        (35, 29),
+    ]
+    test_output_sizes_1 = [
+        # single integer
+        22, 27, 28, 36,
+        # single integer in tuple/list
+        [22, ], (27, ),
+    ]
+    test_output_sizes_2 = [
+        # two integers
+        [22, 22], [22, 28], [22, 36],
+        [27, 22], [36, 22], [28, 28],
+        [28, 37], [37, 27], [37, 37]
+    ]
+
+    for height, width in input_sizes:
+        img = Image.new("RGB", size=(width, height), color=127)
+
+        for osize in test_output_sizes_1:
+            for max_size in (None, 37, 1000):
+
+                t = transforms.Resize(osize, max_size=max_size)
+                result = t(img)
+
+                msg = "{}, {} - {} - {}".format(height, width, osize, max_size)
+                osize = osize[0] if isinstance(osize, (list, tuple)) else osize
+                # If size is an int, smaller edge of the image will be matched to this number.
+                # i.e, if height > width, then image will be rescaled to (size * height / width, size).
+                if height < width:
+                    exp_w, exp_h = (int(osize * width / height), osize)  # (w, h)
+                    if max_size is not None and max_size < exp_w:
+                        exp_w, exp_h = max_size, int(max_size * exp_h / exp_w)
+                    assert result.size == (exp_w, exp_h), msg
+                elif width < height:
+                    exp_w, exp_h = (osize, int(osize * height / width))  # (w, h)
+                    if max_size is not None and max_size < exp_h:
+                        exp_w, exp_h = int(max_size * exp_w / exp_h), max_size
+                    assert result.size == (exp_w, exp_h), msg
+                else:
+                    exp_w, exp_h = (osize, osize)  # (w, h)
+                    if max_size is not None and max_size < osize:
+                        exp_w, exp_h = max_size, max_size
+                    assert result.size == (exp_w, exp_h), msg
+
+    for height, width in input_sizes:
+        img = Image.new("RGB", size=(width, height), color=127)
+
+        for osize in test_output_sizes_2:
+            oheight, owidth = osize
+
+            t = transforms.Resize(osize)
+            result = t(img)
+
+            assert (owidth, oheight) == result.size
+
+    with pytest.warns(UserWarning, match=r"Anti-alias option is always applied for PIL Image input"):
+        t = transforms.Resize(osize, antialias=False)
+        t(img)
```

Comment:
To avoid validating this on every parametrization, this should probably be exported into a single test like `test_resize_antialias_error()`